tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -11166,52 +11166,55 @@ void Sema::DiagnoseReachableHLSLMethodCall(const CXXMethodDecl *MD,
               << MD->getNameAsString() + " with SamplerComparisonState"
               << "6.8";
         }
-      }
+      } else {
```

Comment:
I didn't think this is what we were going to do.  I thought all the checks were going to be limited to the new overload by first checking the intrinsic overload, then if it was the new one, do all the other checks.  Like so:

```
if it's SamplerComparisonTy:
  if < SM 6.8:
    emit diagnostic
  check shader stage
  check threadgroup dimensions for compute-like stages
```

This code looks like it does this:
```
if < SM 6.8:
  if new intrinsic:
    emit diagnostic
else: // SM is 6.8+
  do the rest of the diagnostics for shader stage and threadgroup dimensions
```

With this code, existing intrinsic usage will hit the new diagnostics when compiling to SM 6.8.

Reply:
Updated.